misc/dbt-materialize/dbt/include/materialize/macros/materializations/unit.sql

```diff
@@ -0,0 +1,51 @@
+-- Copyright Materialize, Inc. and contributors. All rights reserved.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License in the LICENSE file at the
+-- root of this repository, or online at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+{%- materialization unit, adapter='materialize' -%}
+
+  {% set relations = [] %}
+
+  {% set expected_rows = config.get('expected_rows') %}
+  {% set expected_sql = config.get('expected_sql') %}
+  {% set tested_expected_column_names = expected_rows[0].keys() if (expected_rows | length) > 0 else get_columns_in_query(sql) %}
+
+  {%- set target_relation = this.incorporate(type='view') -%}
+  {%- set temp_relation = make_temp_relation(target_relation) -%}
+
+  {%- call statement(auto_begin=True) -%}
+    {{ materialize__create_view_as(temp_relation, get_empty_subquery_sql(sql)) }}
```

Comment:
The `dbt-adapter` package by default uses temporary tables for the unit tests:

https://github.com/dbt-labs/dbt-adapters/blob/eda109542efd1703fd08e242006ae4d3fc476b38/dbt/include/global_project/macros/materializations/tests/unit.sql#L11

As we don't support `CREATE TEMPORARY TABLE v AS (SELECT...` I've changed this to just use a view instead which works as expected. But maybe it will be better to extend this to actually use a temporary view instead.

Open to suggestions @morsapaes!

Reply:
I just tried this but it does not work with the temp views out of the box, because of the way that views are being looked up:

```sql
select
          column_name,
          data_type,
          character_maximum_length,
          numeric_precision,
          numeric_scale

      from INFORMATION_SCHEMA.columns
      where table_name = 'test_my_model__dbt_tmp220259074675'
```

I guess that it works fine with temporary tables but we don't seem to keep a record of the temporary views.

However looking into this further, it seems that we should be all fine using regular non-materialized views. We have a clean up step in the unit macro already:
```
{% do adapter.drop_relation(temp_relation) %}
```

Which does `drop view if exists "test_some_model__dbt_tmp215959594739" cascade` after the test is done.

Reply:
My understanding is that the preference to use temporary tables over views is related to performance, and also safeguarding against concurrent invocations. It seems like `create_view_as_sql` doesn't support a `temporary` predicate because most databases lack support for temporary views — we can support it, though. I got carried away and have a version using temporary views half-working locally, but don't want to block this PR on it. Let's address this as follow-up, since the current approach is probably fine due to some extra suffixing `dbt-postgres` does when naming the tables because of [this limitation](https://github.com/MaterializeInc/materialize/issues/15863).